balance/src/lib.rs

```diff
@@ -0,0 +1,351 @@
+extern crate futures;
+extern crate h2;
+extern crate http;
+extern crate tower_balance;
+extern crate tower_h2;
+
+use futures::{Async, Poll};
+use tower_balance::load::Instrument;
+
+/// Instruments HTTP responses to drop handles when their first body message is received.
+#[derive(Clone, Debug, Default)]
+pub struct PendingUntilFirstData(());
+
+/// Instruments HTTP responses to drop handles when their streams completes.
+#[derive(Clone, Debug, Default)]
+pub struct PendingUntilEos(());
+
+/// An instrumented HTTP body that drops its handle when the first data is received.
+pub struct PendingUntilFirstDataBody<T, B> {
```

Comment:
Should these body types impl `Debug` for cases when a body type is bounded with `Debug`?